client/src/modules/blockEditor/actions.ts

```diff
@@ -62,11 +63,63 @@ export const exportSelection = () => {
     const { hoveredItem } = blockEditor;
 
     const block = blocks.blocks.find(b => b.id === hoveredItem.blockId);
+
     const uuid = uuidv1();
     if (block) {
+      const fragment = block.value.fragment;
+      const topLevelNodes = fragment.nodes.toJSON()[0].nodes;
+
+      const onlyOneNodeThatIsPointerExport = nodes => {
+        return (
+          nodes.length === 1
+          &&
+          nodes[0].type === "pointerExport"
+        );
+      };
+
+      const twoNodesFirstExportSecondEmptyString = nodes => {
+        return (
+          nodes.length === 2
+          &&
+          nodes[0].type === "pointerExport"
+          &&
+          nodes[1].object === "text"
+          && nodes[1].leaves[0].text === ""
+        );
+      };
+
+      let nodes = topLevelNodes;
+
+      while (
+        onlyOneNodeThatIsPointerExport(nodes)
+        ||
+        twoNodesFirstExportSecondEmptyString(nodes)
+      ) {
+          nodes = nodes[0].nodes;
+        }
```

Comment:
What's the intention/logic behind the six lines above?

Reply:
Basically trying to get to the inner-nested nodes of a fragment. I added an explanatory comment here: https://github.com/oughtinc/mosaic/pull/145/commits/529628930bf22b2a36fa72d37e6d81dfa3d62c0c

I'll think if there might be a simpler way to approach this.

Reply:
How can I see that the cases are the only relevant ones, i.e. how do I know that things like [empty string, export] or [export, empty string, empty string] aren't possible?

Reply:
Typically, when there is more than just the exported pointer, you actually want to include the exported pointer in the content of what you're exporting (thus creating a nested export). The [export, empty string] that I'm currently looking for is an edge case when the text you've selected goes all the way to the end of the pointer. For example, suppose you've exported the content `AAABBB`, and now you want to create a nested pointer that just encompasses the `BBB`. When the expanded selection encompasses `BBB`, the end of the selection is actually at the start of the next node. If were selecting `AAA`, the start of the selection is not at the end of the previous node. So with cursor selection, there is an asymmetry between the start and end.

This actually interacts with the spacer characters in a way that I'm currently addressing.

Reply:
OK the logic here is a bit different now: https://github.com/oughtinc/mosaic/pull/145/commits/b16d2070776d6cdc40e3942a503b28fca30e4f3a

What I'm doing is first checking to see whether we're already in a pointer when we're trying to export. If we are, then I check to see whether the selection is up against an edge. If it is, then I move it over so that it's no longer up against an edge. This means I can now ignore the possibility where we are worried about [export, empty string], since that is a possibility that only arises when the selection extends to the end of the pointer.